crates/sui-core/src/consensus_handler.rs

```diff
@@ -646,238 +740,1639 @@ impl<C: CheckpointServiceNotify + Send + Sync> ConsensusHandler<C> {
-                .consensus_bad_nodes_stake_threshold(),
+            protocol_config.consensus_bad_nodes_stake_threshold(),
         );
 
+        // DONE(commit-handler-rewrite): update metrics
         self.metrics
             .consensus_committed_subdags
             .with_label_values(&[&leader_author.to_string()])
             .inc();
 
+        let mut state = CommitHandlerState {
+            output: ConsensusCommitOutput::new(commit_info.round),
+            dkg_failed: false,
+            randomness_round: None,
+            indirect_state_observer: Some(IndirectStateObserver::new()),
+            initial_reconfig_state: self
+                .epoch_store
+                .get_reconfig_state_read_lock_guard()
+                .clone(),
+        };
+
+        // DONE(commit-handler-rewrite): update transaction status (rejected/finalized) and update metrics
+        let transactions = self.filter_consensus_txns(
+            state.initial_reconfig_state.clone(),
+            &commit_info,
+            &consensus_commit,
+        );
+        // DONE(commit-handler-rewrite): de-duplicate transactions
+        let transactions = self.deduplicate_consensus_txns(&mut state, &commit_info, transactions);
+
+        let mut randomness_manager = self.init_randomness(&mut state, &commit_info);
+
+        // DONE(commit-handler-rewrite): Split transactions into different types for processing.
+        let CommitHandlerInput {
+            user_transactions,
+            capability_notifications,
+            execution_time_observations,
+            checkpoint_signature_messages,
+            randomness_dkg_messages,
+            randomness_dkg_confirmations,
+            end_of_publish_transactions,
+            new_jwks,
+        } = self.build_commit_handler_input(transactions);
+
+        self.process_jwks(&mut state, &commit_info, new_jwks);
+        self.process_capability_notifications(capability_notifications);
+        self.process_execution_time_observations(&mut state, execution_time_observations);
+        self.process_checkpoint_signature_messages(checkpoint_signature_messages);
+
+        let randomness_dkg_updates = self.process_randomness_dkg_messages(
+            randomness_manager.as_deref_mut(),
+            randomness_dkg_messages,
+        );
+
+        let randomness_dkg_confirmation_updates = self.process_randomness_dkg_confirmations(
+            &mut state,
+            randomness_manager.as_deref_mut(),
+            randomness_dkg_confirmations,
+        );
+
+        // DONE(commit-handler-rewrite): [ssm] advance randomness state if needed
+        if randomness_dkg_updates || randomness_dkg_confirmation_updates {
+            if let Some(randomness_manager) = randomness_manager.as_mut() {
+                randomness_manager
+                    .advance_dkg(&mut state.output, commit_info.round)
+                    .await
+                    .expect("epoch ended");
+            }
+        }
```

Comment:
Should this code block be inside `process_randomness_dkg_confirmations`?

Reply:
move all the dkg code into its own function, much cleaner